src/shared/varlink-idl.c

```diff
@@ -1004,6 +1004,9 @@ int varlink_idl_parse(
                 case STATE_METHOD_ARROW:
                         assert(symbol);
 
+                        if (!token)
+                                return log_debug_errno(SYNTHETIC_ERRNO(EBADMSG), "%u:%u: Premature EOF.", *line, *column);
```

Comment:
can you add this to the STATE_METHOD thing above too? and actually STATE_TYPE, STATE_ERROR below too.

seems i forgot that not just once but four times